test/scripts/helpers/list_tags.js

```diff
@@ -205,3 +205,41 @@ describe('list_tags', () => {
     ].join(''));
   });
 });
+
+describe('list_tags transform', () => {
+  const Hexo = require('../../../lib/hexo');
+  const hexo = new Hexo(__dirname);
+  const Post = hexo.model('Post');
+
+  const ctx = {
+    config: hexo.config
+  };
+
+  const listTags = require('../../../lib/plugins/helper/list_tags').bind(ctx);
+
+  before(async () => {
+    await hexo.init();
+    const posts = await Post.insert([
+      {source: 'foo', slug: 'foo'}
+    ]);
+
+    // TODO: Warehouse needs to add a mutex lock when writing data to avoid data sync problem
```

Comment:
See also https://github.com/hexojs/hexo/issues/4430

Reply:
I just copied this part from the tests above. Not sure how to fix it

Reply:
I just add that comment to add a link back to the already opened issue (for later fix). The current workaround already looks good for this PR.